activerecord/lib/active_record/connection_adapters/abstract/database_statements.rb

```diff
@@ -524,12 +520,21 @@ def combine_multi_statements(total_sql)
         end
 
         # Returns an ActiveRecord::Result instance.
-        def select(sql, name = nil, binds = [])
-          exec_query(sql, name, binds, prepare: false)
-        end
+        def select(sql, name = nil, binds = [], prepare: false, async: false)
```

Comment:
In all other method signatures of this PR `async` is a boolean, here it can also be a `FutureResult::SelectAll`. Would something like`future_class` be more clear?